File: Semester 3/OOP/Week2/Linear.java
import java.util.Scanner;

class Linear
{
	public static void main(String args[])
	{
		int n, i, j,val,a[];
		System.out.print("Enter n: ");
		Scanner sc = new Scanner(System.in);
		n = sc.nextInt();
		a = new int[n];
		System.out.print("Enter elements: ");
		for (i=0; i<n; i++) 
		{
			a[i]=sc.nextInt();
		}
		System.out.print("Enter value to be searched: ");
		val = sc.nextInt();
		lsearch(n, a,val);
	}
	static void lsearch(int n, int a[], int val)
	{
		int i;
		for (i=0; i<n; i++)
		{
			if (a[i]==val) break;
		}
		if (i<n) System.out.print("Element was found at position " + (i+1));
		else System.out.print("Element not found");
	}
}

File: Semester 3/DSA/Week01/linear.c
#include <stdio.h>

int lsearch(int value, int a[100], int n)
{
	int i;
	for (i=0; i<n; i++)
	{
		if (a[i]==value) 
		{
			return i;
		}
	}
	return -1;
}

int main()
{
	int i, n, a[100], index, value;
	printf("Enter number of elements: ");
	scanf("%d", &n);
	printf("Enter elements: ");
	for (i=0; i<n; i++)
	{
		scanf("%d", &a[i]);
	}	
	printf("Enter value to be searched: ");
	scanf("%d", &value);
	index = lsearch(value, a, n);
	printf("%d is at index %d", value, index);
}